lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1790,16 +1841,21 @@ class LateSSAReplacement : public FuncOpPartialLoweringPattern {
 
   LogicalResult PartiallyLowerFuncToComp(mlir::FuncOp funcOp,
                                          PatternRewriter &) const override {
-    funcOp.walk([&](scf::WhileOp whileOp) {
+    auto whileReplacement = [&](Operation *op) {
       /// The yielded values returned from the while op will be present in the
       /// iterargs registers post execution of the loop.
       /// This is done now, as opposed to during BuildWhileGroups since if the
       /// results of the whileOp were replaced before
       /// BuildOpGroups/BuildControl, the whileOp would get dead-code
       /// eliminated.
+      WhileOpInterface whileOp(op);
       for (auto res : getComponentState().getWhileIterRegs(whileOp))
-        whileOp.getResults()[res.first].replaceAllUsesWith(res.second.out());
-    });
+        whileOp.getOperation()->getResults()[res.first].replaceAllUsesWith(
+            res.second.out());
+    };
+
+    funcOp.walk([&](scf::WhileOp op) { whileReplacement(op); });
+    funcOp.walk([&](staticlogic::PipelineWhileOp op) { whileReplacement(op); });
```

Comment:
Suggesting to move op type checking into `whileReplacement` to avoid walking the IR twice.

Reply:
After working a little further on supporting PipelineWhileOp, it will need to handle this differently, so I'll just put back the original handling for WhileOp in this first patch.